BuildResidentialHPXML/resources/geometry.rb

```diff
@@ -1278,9 +1278,44 @@ def self.add_windows_to_wall(surface, window_area, window_gap_y, window_gap_x, w
     window_width = Math.sqrt((window_area / num_windows.to_f) / window_aspect_ratio)
     window_height = (window_area / num_windows.to_f) / window_width
     width_for_windows = window_width * num_windows.to_f + window_gap_x * num_window_gaps.to_f
+
     if width_for_windows > wall_width
-      runner.registerError("Could not fit windows on #{surface.name}.")
-      return false
+      surface_area = UnitConversions.convert(surface.grossArea, 'm^2', 'ft^2')
+      runner.registerWarning("Surface #{surface.nameString} has an area of #{surface_area}, params: window_area=#{window_area} ft^2, window_gap_y=#{window_gap_y} ft, window_gap_x=#{window_gap_x} ft, window_aspect_ratio=#{window_aspect_ratio}, max_single_window_area=#{max_single_window_area} ft^2, facade=#{facade}")
+      runner.registerWarning("Could not fit windows on #{surface.name}. Fall back to WWR")
+      wwr = window_area / surface_area
+      if wwr >= 0.90
+        wwr = 0.90
+      end
```

Comment:
My reaction to this is:
1. We probably don't need a warning just because we are relaxing constraints around aspect ratio, window gap, etc. to fit the window area. If we are preserving the window area, the user is probably fine.
2. But we _do_ need a warning if we are going to reduce the window area.

So how about this instead?
```suggestion
      wwr = window_area / surface_area
      if wwr > 0.90
        runner.registerWarning("Could not fit windows on #{surface.name}; reducing window area to 90% WWR.")
        wwr = 0.90
      end
```

Reply:
I'm made the change somewhere else and got the exact same feedback from someone else, so I guess we do need a warning :)

I didn't add it on purpose originally:

* Either you don't care, because it has no impact overall since you managed to add more window area to another wall for that facade
* Or you do care, but you'll get https://github.com/NREL/OpenStudio-HPXML/blob/9528831dd85ec80027ec93e0d994a7a76f86ea4f/BuildResidentialHPXML/resources/geometry.rb#L1151